code/game/objects/items/devices/radio/detpack.dm

```diff
@@ -0,0 +1,323 @@
+/obj/item/device/radio/detpack
+	name = "detonation pack"
+	desc = "Programmable remotely triggered 'smart' explosive used for demolitions and impromptu booby traps. Can be set to breach or demolition detonation patterns."
+	gender = PLURAL
+	icon = 'icons/obj/det.dmi'
+	icon_state = "detpack_off"
+	item_state = "plasticx"
+	flags_item = NOBLUDGEON
+	frequency = 1457
+	w_class = 2.0
+	origin_tech = "syndicate=2"
+	on = FALSE
+	layer = MOB_LAYER - 0.1
+	var/armed = FALSE
+	var/timer = 10.0
+	var/code = 2
+	var/det_mode = FALSE //FALSE for breach, TRUE for demolition.
+	var/atom/plant_target = null //which atom the detpack is planted on
+	var/target_drag_delay = null //store this for restoration later
+
+/obj/item/device/radio/detpack/examine(mob/user)
+	. = ..()
+	var/details
+	if(timer)
+		details += "Its timer has [timer] seconds left."
+	if(det_mode)
+		details += " It appears set to demolition mode."

```

Comment:
If timer is a falsey value you'll be starting the string with a space. It's both more performant and better functionaly to do this with a list.Join()

```
var/list/details = list()
if(timer)
 details += "Its timer has [timer] seconds left."
...
to_chat(user, "<span class='warning'>[details.Join(" ")]</span>")
```
Yes it really is more performant to do string operations like this over `+`, byond's string modifications are notoriously bad